packages/@react-stately/data/src/useListData.ts

```diff
@@ -123,9 +123,9 @@ export interface ListData<T> {
   /**
    * Updates an item in the list.
    * @param key - The key of the item to update.
-   * @param newValue - The new value for the item.
+   * @param newValue - The new value for the item, or a function that returns the new value based on the previous value.
    */
-  update(key: Key, newValue: T): void
+  update(key: Key, newValue: T | ((prev: T) => T)): void
```

Comment:
This works like useState's setState callback, we already had the previous value, so it was trivial to expose and it simplified my logic in my examples, no longer have to track the previous list state and depend on renders